tools/qlog/src/events/quic.rs

```diff
@@ -200,10 +201,19 @@ pub enum StreamState {
 }
 
 #[derive(Serialize, Deserialize, Clone, PartialEq, Debug)]
-#[serde(rename_all = "snake_case")]
 pub enum ErrorSpace {
-    TransportError,
-    ApplicationError,
+    #[serde(alias = "transport_error")]
+    Transport,
+    #[serde(alias = "application_error")]
+    Application,
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Debug)]
+#[serde(untagged)]
+pub enum ErrorCode {
+    TransportError(TransportError),
+    ApplicationError(ApplicationError),
```

Comment:
It seems a bit weird that the `quic` module references an `h3` type. Maybe `ErrorCode` and `ErrorSpace` should go in `mod.rs` instead?

Reply:
good spot! The current code models the qlog datamodel and `ApplicationError` is defined in the H3 spec, I agree it feels wrong but I'd say that it is probably a bug in the spec. To maintain consitency we could leave it as is now, and fixup when qlog changes?

see https://quicwg.org/qlog/draft-ietf-quic-qlog-h3-events.html#name-applicationerror and https://quicwg.org/qlog/draft-ietf-quic-qlog-quic-events.html#name-connectioncloseframe

Reply:
made https://github.com/quicwg/qlog/issues/183

Reply:
I've dropped this commit from the PR because there isn't an easy resolution right now, we can fix it seperately pending the outcome